Sources/XCTest/Public/XCTestCase.swift

```diff
@@ -114,9 +135,46 @@ open class XCTestCase: XCTest {
         self.testClosure = testClosure
     }
 
+    #if USE_SWIFT_CONCURRENCY_WAITER
+    internal func _invokeTestAsync() async {
+        await performSetUpSequence()
+
+        do {
+            if skip == nil {
+                try testClosure(self)
```

Comment:
It's very important for behavioral consistency in this mode that when calling out to *non-*`async` user-provided code (test methods, and `setUp` and `tearDown` overrides) that those synchronous functions run on the main actor. I'm not certain, but I don't think that invariant is maintained by the current code. Can you comment on that?

(This was something that we struggled with a bit, and initially got wrong when we were working on support for `async` test methods a while back in https://github.com/apple/swift-corelibs-xctest/pull/326)

Reply:
I think we can keep the consistency by adding `@MainActor` on this method.